tests/net_dynamic_hb.rs

```diff
@@ -503,16 +503,22 @@ where
             "the network is already captured by the faulty nodes"
         );
 
-        let max_number_correct_nodes_for_remove = n - (f * 3 + 1);
-        let remove_from_correct = rng.gen_range(0, max_number_correct_nodes_for_remove + 1);
-        let remove_from_faulty = if remove_from_correct == 0 && f > 0 {
-            // if we can't remove any correct node, because network will become in an
-            // incorrect state we have to remove at least 1 faulty node
+        let max_number_correct_nodes_for_remove = |n: usize, f: usize| n - (f * 3 + 1);
+
+        let remove_from_faulty = if max_number_correct_nodes_for_remove(n, f) == 0 && f > 0 {
+            // we can't remove any correct node now, because network will become in an incorrect state
+            // but we will be able to remove correct nodes after removing several malicious nodes,
+            // so at least one malicious node should be guaranteed removed
             rng.gen_range(1, f + 1)
         } else {
             rng.gen_range(0, f + 1)
         };
 
+        let new_f = f - remove_from_faulty;
+        let new_n = n - remove_from_faulty;
+        let remove_from_correct =
+            rng.gen_range(0, max_number_correct_nodes_for_remove(new_n, new_f) + 1);
+
```

Comment:
This all seems unnecessarily complicated to me. If I understand correctly, we want to:
* remove at least one node if `n > 1`, and
* make sure `new_n > 3 * new_f`.

So why not just:
```rust
if n <= 1 {
    return BTreeSet::new(); // Only one node left; nothing to remove.
}
let new_n = rng.gen_range(1, n);
let new_f = rng.gen_range(0, f.min(util::max_faulty(new_n)) + 1);
```

Reply:
Ok. It really is a bit easier.